src/lib/tls/msg_client_hello.cpp

```diff
@@ -292,14 +281,26 @@ Client_Hello::Client_Hello(const std::vector<uint8_t>& buf)
    const uint8_t minor_version = reader.get_byte();
 
    m_version = Protocol_Version(major_version, minor_version);
-
    m_random = reader.get_fixed<uint8_t>(32);
-
    m_session_id = reader.get_range<uint8_t>(1, 0, 32);
 
+   std::unique_ptr<HashFunction> sha256;
+   if(m_version.is_datagram_protocol())
```

Comment:
why not merge this if with the if in line 294 and the if in line 297?

Reply:
Thanks that approach is much clearer I do not remember why I wrote it the way I did.